tests/builder/Data/ByteString/Builder/Tests.hs

```diff
@@ -98,10 +98,19 @@ testHandlePutBuilder =
     testProperty "hPutBuilder" testRecipe
   where
     testRecipe :: (UnicodeString, UnicodeString, UnicodeString, Recipe) -> Bool
-    testRecipe args@(UnicodeString before,
-                     UnicodeString between,
-                     UnicodeString after, recipe) =
+    testRecipe args =
       unsafePerformIO $ do
+        let (UnicodeString a1, UnicodeString a2, UnicodeString a3, recipe) = args
+#if MIN_VERSION_base(4,5,0)
+            before  = a1
+            between = a2
+            after   = a3
+#else
+            safeChr = \c -> c < '\xEF00' || c > '\xEFFF'
+            before  = filter safeChr a1
+            between = filter safeChr a2
+            after   = filter safeChr a3
```

Comment:
Could you add a comment or just a reference to the issue?

Reply:
Sure, makes sense. Will do shortly.